admin/class-convertkit-admin-restrict-content.php

```diff
@@ -172,4 +221,46 @@ public function maybe_display_restrict_content_post_state( $post_states, $post )
 
 	}
 
+	/**
+	 * Determines if the current request is for a WP_List_Table, and if so that
+	 * the Post Type we're viewing supports Restrict Content functionality.
+	 *
+	 * @since   2.1.0
+	 *
+	 * @return  bool    Is WP_List_Table request for a supported Post Type.
+	 */
+	private function is_wp_list_table_request_for_supported_post_type() {
+
+		// Bail if we cannot determine the screen.
+		if ( ! function_exists( 'get_current_screen' ) ) {
+			return false;
+		}
+
+		// Get screen.
+		$screen = get_current_screen();
+
+		// Bail if we're not on an edit.php screen.
+		if ( $screen->base !== 'edit' ) {
+			return false;
+		}
+
+		// Return whether Post Type is supported for Restrict Content functionality.
+		return in_array( $screen->post_type, convertkit_get_supported_restrict_content_post_types(), true );
+
+	}
+
+	/**
+	 * Get the current post type based on the screen that is viewed.
+	 *
+	 * @since   2.1.0
+	 *
+	 * @return  string
+	 */
+	private function get_current_post_type() {
+
+		$screen = get_current_screen();
```

Comment:
Do we need the bailout like on line 235?

Reply:
Absolutely, thanks for spotting this.